fe/src/utils/category.ts

```diff
@@ -0,0 +1,5 @@
+// @todo - Update this or remove after back-end implementation.
```

Comment:
I think this can be removed now?

Reply:
@Orzelius This seems kinda frustrating atm.  We could either use this custom type, or use no type at all [here](https://github.com/TPT-Loane/TPT-Loane/pull/108/files#diff-19e306ac23550d961cc0a799e8e282e7661c4ae12fd92638652a6419a72f13fdR17) (both technically work). 

The type generated by apollo has some specific property types set and couldn't be used right away. 🤔 

![type](https://user-images.githubusercontent.com/48998862/119984855-92ac5a80-bfca-11eb-8401-8937e0434b2a.JPG)
![typeerror](https://user-images.githubusercontent.com/48998862/119984856-9344f100-bfca-11eb-9a74-e34da7549469.JPG)

Reply:
Currently i just removed it, see whatca think.

Reply:
We should only used gen-types. You can just return error or some kind of loading indicator if the data is undefined.

Reply:
```tsx
  const categories = categoriesQuery.data && categoriesQuery.data.categories;
  if (!categories) return <LoaderOrSmthng />
  categories.map(c => c); // <- this works becuase categories is no longer `| undefined`
```

Reply:
Currently im just returning null if no categories are found, since we only display categories in dropdown atm. 

But for future reference i think some universal loading indicator component could be added to whereever we might need them. Actually a [spinner](https://chakra-ui.com/docs/feedback/spinner) is already provided by chakra. 🤔